lib/ui/hooks.dart

```diff
@@ -161,7 +161,9 @@ void _dispatchPlatformMessage(String name, ByteData data, int responseId) {
       },
     );
   } else {
-    window._respondToPlatformMessage(responseId, null);
+    channelBuffers.push(name, data, (ByteData responseData) {
```

Comment:
Related to the comment on the other PR. I think there are 2 lifecycles and we can't really conflate them. There's setting the onPlatformMessage which happens once per binding, and there's adding per channel handlers which happens at arbitrary time. 

Once the onPlatformMessage is attached, all buffers will stop queueing. But there are no guarantee that the following sequence of actions won't happen. 

1- run entrypoint, entrypoint runApps, which attaches window.onPlatformMessage. 
2- engine sends a message on some channel. It hits _DefaultBinaryMessenger. _handlers for that channel is null. Nothing happens. It's also not queued since onPlatformMessage is not null.
3- framework side code creates that channel and attaches the setMessageHandler for that channel. Nothing happens. 

It probably works now since the synchronous execution of runApp+binding constructor+our system channels setting handler all happens at the same time. But we can't guarantee that for users' own channels.

Reply:
I've addressed this in the other PR.  Both cases get queued up now.